### PR TITLE
Implement Hybrid Mode and Image Summary

### PR DESCRIPTION
**Pull request by Cadair**
This implements a hybrid mode where you can first compare against a (local) hash library and then if that fails fall back to comparing against a (remote) baseline image set.

based on #100 which is based on #98 which was based on #97

### CODE REVIEW

File: pytest_mpl/plugin.py [hunk 1]
```diff
@@ -109,10 +147,15 @@ def pytest_addoption(parser):
     group.addoption('--mpl-hash-library',
                     help="json library of image hashes, relative to "
                     "location where py.test is run", action='store')
+    group.addoption('--mpl-generate-summary', action='store_true',
```
Comment (astrofrog) [inline pytest_mpl/plugin.py:150]: I wonder if we want to allow for non-HTML summaries in future. Can we easily extend this to allow e.g. ``--mpl-generate-summary=xml`` and have the one without a format default to HTML? (if so then we can extend this in a back-compat way in future)

**Review by astrofrog — COMMENTED** [review]
Just a small comment below - and also this needs some docs in the README.

File: pytest_mpl/plugin.py (hunk 1 above)
Comment (Cadair) [reply pytest_mpl/plugin.py:150]: This is actually a good call, I was thinking of a json summary with a fancy (static) html / js frontend

Comment (astrofrog) [reply pytest_mpl/plugin.py]: I think the description needs updating to mention that the format should be given? Also maybe validate the formats?

**Review by astrofrog — APPROVED** [review]
Looks good and work nicely - the HTML page can probably be made prettier but this is fine as-is for now :)

**Pull request merged**